geoprocessing/src/modules/shapefiles/shapefiles.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Logger } from '@nestjs/common';
+
+import { createReadStream, rmdirSync } from 'fs';
+const mapshaper = require('mapshaper');
+const unzipper = require('unzipper');
+
+@Injectable()
+export class ShapeFileService {
+  private readonly logger: Logger = new Logger(ShapeFileService.name);
+  constructor() {}
+
+  private _unzipShapefile(fileInfo: Express.Multer.File): Promise<string> {
+    return new Promise((resolve, reject) => {
+      createReadStream(fileInfo.path)
+        .pipe(
+          unzipper.Extract({
+            path: fileInfo.destination + fileInfo.filename.replace('.zip', ''),
+          }),
+        )
+        .on('close', () =>
+          resolve(`${fileInfo.filename} extracted succesfully`),
+        )
+        .on('error', () =>
+          reject(new Error(`${fileInfo.filename} could not been extracted`)),
+        );
+    });
+  }
+
+  private async _shapeFileToGeoJson(fileInfo: Express.Multer.File) {
+    const outputKey = `shapefile-${new Date().getTime()}.gejson`;
+    const _geoJson = await mapshaper.applyCommands(
+      `-i ${fileInfo.path.replace('.zip', '')}/*.shp -o ${outputKey}`,
+    );
+
+    return JSON.parse(_geoJson[outputKey].toString('utf-8'));
+  }
+
+  private wipeOutFolder(path: string): void {
```

Comment:
very tiny note on naming (as usual from me :grin:) - is this wipe-out folder, or wipe the output folder? I would just use "remove", "delete" or "unlink" folder.

but besides naming (which you can ignore - I just commented because wipe triggers images of bit flipping to remove traces of whatever was written to disk!) - this function is basically a thin wrapper around `rmdirSync()`, so also taking into account my other comment below, I would suggest to make this a sort of shared/utility function for "removing folder with temporary artifacts for job X", assuming that we can stick to a consistent handling of paths for various kinds of file-related jobs and therefore the temp directories we may create to write source files, unzip them where applicable, process them, read results from, etc.